src/View/BookingCodes.php

```diff
@@ -76,4 +373,68 @@ public static function renderCSV( $timeframeId = null ) {
 		die;
 	}
 
+    /**
+     * action for sending Booking Codes by E-mail
+     * 
+     * @param int $timeframeId      ID of requested timeframe
+     * @param int $tsFrom           Timestamp of first Booking Code
+     * @param int $tsTo             Timestamp of last Booking Code
+     */
+    public static function emailCodes($timeframeId = null, $tsFrom=null, $tsTo=null) { 
```

Comment:
Das passt vielleicht auch nicht so gut in die View Klasse, oder?

Reply:
Hast du einen Vorschlag wohin damit? Es ist die Implementierung der custom Wordpress Action und wird direkt über die Links im Timeframe getriggert. Eventuell in der Messages/BookingCodesMessage.php ?

Reply:
@printpagestopdf @hansmorb 

Ein Vorschlag würde mich auch interessieren, was spricht denn gegen den Standort @hansmorb hier? Die bisherigen custom action zum Erzeugen der CSV `renderCSV` liegt auch in der View. Nur die anderen Message-Actions die über den Scheduler gehen haben eine Methode in einem Service. Genau wie Service\BookingCodes ja auch über den Scheduler geht. Und der Code im Messages Package ist doch eigentlich immer nur API (z.B. in Services) und implementiert nie eine Action?

Reply:
Meinem Verständnis nach ist die View Klasse eher nur für Sachen, die in das Frontend gerendert werden, wie die renderCSV, die ja auch mehr Rendering direkt übernimmt. Oder halt Funktionen deren übergebener Wert direkt ausgegeben wird. Deshalb glaube ich, dass die besser in den Service passt.